### PR TITLE
Refactor: extract git logic from install flux

### PR DESCRIPTION
**Pull request by martina-if**
Extract git logic from install flux to another package to make code testable.

- [x] Code compiles correctly (i.e `make build`)
- [x] Added tests that cover your change (if possible)
- [x] All unit tests passing (i.e. `make test`)
- [x] Added/modified documentation as required (such as the `README.md`, and `examples` directory)
- [x] Manually tested

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -17,7 +17,7 @@ require (
 	github.com/coreos/go-systemd v0.0.0-20190620071333-e64a0ec8b42a // indirect
 	github.com/dave/jennifer v1.3.0
 	github.com/dlespiau/kube-test-harness v0.0.0-20190110151726-c51c87635b61
-	github.com/docker/docker v1.13.1 // indirect
+	github.com/docker/docker v1.13.1
```
Comment (errordeveloper) [inline go.mod]: Did you figure out why did this change?

Comment (martina-if) [reply go.mod]: no :woman_shrugging:

**Comment by errordeveloper** [comment]
@martina-if is this complete or you want to change something?

**Comment by martina-if** [comment]
@errordeveloper this step is complete so I think the PR is good to go

File: pkg/ctl/install/install.go [hunk 2]
```diff
@@ -138,12 +140,15 @@ func newFluxInstaller(rc *cmdutils.ResourceCmd, opts *installFluxOpts) (*fluxIns
 	if err != nil {
 		return nil, errors.Errorf("cannot create Kubernetes client set: %s", err)
 	}
+
+	gitClient := git.NewGitClient(ctx, opts.templateParams.GitUser, opts.templateParams.GitEmail, opts.timeout)
```
Comment (marccarre) [inline pkg/ctl/install/install.go]: Nit: we could pass a `GitClientParams` (or so) `struct` to group all git-related configuration & options.

File: pkg/ctl/install/install.go.orig [hunk 3]
```diff
@@ -0,0 +1,657 @@
...
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+<<<<<<< HEAD
```
Comment (marccarre) [inline pkg/ctl/install/install.go.orig]: Is this `.orig` file an accidental addition? This looks like the output of git during some conflict resolution, so I'm guessing we don't want it here, do we?

Comment (martina-if) [reply pkg/ctl/install/install.go.orig]: upssss thanks!

File: pkg/ctl/install/install.go (hunk 2 above)
Comment (martina-if) [reply pkg/ctl/install/install.go]: since they are passed only once to a constructor of the git.Client struct I will leave them for now

**Comment by martina-if** [comment]
Approved by @marccarre via slack

**Pull request merged**